torch/utils/data/distributed.py

```diff
@@ -37,12 +38,17 @@ def __init__(self, dataset, num_replicas=None, rank=None):
         self.epoch = 0
         self.num_samples = int(math.ceil(len(self.dataset) * 1.0 / self.num_replicas))
         self.total_size = self.num_samples * self.num_replicas
+        self.shuffle = shuffle
 
     def __iter__(self):
         # deterministically shuffle based on epoch
         g = torch.Generator()
         g.manual_seed(self.epoch)
-        indices = torch.randperm(len(self.dataset), generator=g).tolist()
+        indices = (
+            torch.randperm(len(self.dataset), generator=g).tolist()
```

Comment:
break this out into a separate if/else block, instead inlining if/else inside indices = (